openmdao/core/tests/test_impl_comp.py

```diff
@@ -363,6 +364,32 @@ def test_list_residuals(self):
         self.assertEqual(text.count('value'), 0)
         self.assertEqual(text.count('resids'), 1)
 
+    def test_list_residuals_with_tol(self):
+        prob = om.Problem()
+        model = prob.model
+
+        model.add_subsystem('p1', om.IndepVarComp('x', 1.0))
+        model.add_subsystem('d1', SellarImplicitDis1())
+        model.add_subsystem('d2', SellarImplicitDis2())
+        model.connect('d1.y1', 'd2.y1')
+        model.connect('d2.y2', 'd1.y2')
+
+        model.nonlinear_solver = om.NewtonSolver(solve_subsystems=False)
+        model.nonlinear_solver.options['maxiter'] = 5
+        model.linear_solver = om.ScipyKrylov()
+        model.linear_solver.precon = om.LinearBlockGS()
+
+        prob.setup()
+        prob.set_solver_print(level=-1)
+
+        prob.run_model()
+
+        stdout = StringIO()
+        outputs = model.list_outputs(residuals_tol=0.01, residuals=True, out_stream=stdout)
+        text = stdout.getvalue().split('\n')
+        self.assertEqual(text[14], 'd1')
+        self.assertEqual(text[16], 'd2')
+
```

Comment:
can you explain how this test works? 
it looks like you end up with d1 and d2 in the result. How do you make sure that anything actually got filtered?

Reply:
`p1` has a resid of 0 and that gets filtered out. `d1` and `d2` have a resid higher than 0.01 so they don't get filtered out.

Reply:
ah, k. Thanks!

Reply:
good use of IVC there :)